File: YooperGreensApp.Core/DomainService/IRepository.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace YooperGreensApp.Core.DomainService
{
    public interface IRepository<T, K>
    {
        //Create Data
        T Create(T item);

        //Read Data
        T FindById(K id);
        IEnumerable<T> ReadAll(IFilterData filter = null);

        //Update Data
        T Update(T item);

        //Delete Data
        T Delete(T item);
        int Count();
    }
}

File: YooperGreens.UI/YooperGreens.UI/wwwroot/Script/Core/RepositoryService/IRepository.ts
module GScope {
    export module RepositoryService {
        export interface IRepository<T, K> {
            Create(item: T): T;
            FindById(id: K): T;
            ReadAll(): T[];
            Update(item: T): T;
            Delete(item: T): T;
            Count(): number;
        }
    }
}